enumerable.rb

```diff
@@ -0,0 +1,112 @@
+module Enumerable
+  def my_each
+    i = 0
+    return enum unless block_given?
+
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    return enum unless block_given?
+
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    return enum unless block_given?
+
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
```

Comment:
- [ ] `my_select` returns an Enumerator if no block is given. Currently, its throwing the error below.
![image](https://user-images.githubusercontent.com/64360974/113575666-e6cc3b80-9626-11eb-869d-31e849d5c93a.png)